### ISSUE TITLE
R Package

### ISSUE THREAD

**Issue by adletaw** [jae0/ecomod#3]
I read that you opted out of creating an R package for this repository, but I wanted to give you the soft sell for doing it anyways.  It is possible to create an R package that you don't intend to put on R-cran and still use the Git version control system to keep it constantly updated.  Once the package structure is created, you would not have to do anything extra - just push changes to Github as usual.  This would make it easier for people trying to get started with these functions.  The process of defining "ecomod.directory" etc. is confusing for people with rougher R skills who have never used the `.Rprofile` file.

Further, there is a really nice package out there for installing packages directly from Github called [devtools](https://github.com/hadley/devtools).  You can use the function `install_github` to do it, and it makes it really easy to keep up with packages that are in development on Github.

In any case, thank you so much for putting your work up free and open source.  If you do nothing else, that is already a great contribution!

**Comment by jae0** [comment]
Hi there. Thanks for the comments. Rpackages are not yet ruled out. Some of
this has matured enough to qualify and we probably will do as you suggest.
Just finding the time is the issue of course..

I will look into the errors and correct them. Thanks!

Best regards,
Jae

**Comment by adletaw** [comment]
Yes of course, finding the time is a big issue!  Good luck.  :smile:

**Comment by jae0** [comment]
Thanks.

WRT your error message. It is simply due to these global variables being
required to correctly bootstrap the system. To do this, simply add the
following three statements to your .Rprofile


ecomod.directory = file.path( "C:", "path", "to", "ecomod" )           ###
replace with correct path to local ecomod directory

ecomod.datadirectory = file.path( "C:", "path", "to", "ecomod_data" )  ###
replace with correct path to local data directory

source( file.path( ecomod.directory, "_ecomodSetup", "ecomod.rprofile.r" )
)

**Comment by adletaw** [comment]
Actually, I had already defined the variables `ecomod.directory` and `ecomod.datadirectory`.  The problem is with the variable `ecomod.workdirectory` which is defined in the script `ecomod.profile.r`.  I resolved the problem by changing from `is.null` to `!exists` (see #2).  Then the script is able to correctly define the `ecomod.workdirectory` variable.  

`is.null` will not work for your purposes because it gives an error message if the object does not exist.  In this case, `ecomod.workdirectory` does not exist until it is defined inside the script `ecomod.rprofile.r`.
